kotlinpoet/src/main/java/com/squareup/kotlinpoet/CodeWriter.kt

```diff
@@ -531,6 +531,20 @@ internal class CodeWriter constructor(
     return null
   }
 
+  /**
+   * Returns the qualified name of the `simpleName` enum constant if it's in the current scope
+   */
+  private fun resolveEnumConstant(simpleName: String): String? {
+    if (typeSpecStack.size > 0) {
+      val typeSpec = typeSpecStack[0]
+      if (typeSpec.isEnum && typeSpec.enumConstants.keys.contains(simpleName)) {
+        return "$packageName.${typeSpec.name}.$simpleName"
```

Comment:
This is technically compatible with the structure of `ClassName`, would there be any undesired side effects (e.g. unwanted imports) if we returned `ClassName` here?

Reply:
I think that'd work. Especially since `resolve()` is very local to this class the risk of side effect seems pretty limited. I'll try this

Reply:
Code updated. Let me know what you think